### PR TITLE
docs: installing section

### LINKED ISSUE #1203: Documentation contains nothing about downloading and installation
The [current documentation](http://pybind11.readthedocs.io/en/latest/) contains absolutely nothing about where to download and how to install the software. It's not that hard to trace things back here to GitHub, but a more explicit pointer would be a good idea! (For those of us on  macOS a comment that [it's available](https://github.com/Homebrew/homebrew-core/blob/master/Formula/pybind11.rb) via [homebrew](https://brew.sh) might also be appropriate.

### PR DESCRIPTION
**Pull request by henryiii**
This creates an installing section.

Closes #1203.

### CODE REVIEW

**Review by bstaletic — APPROVED** [review]
Just one thought. Is there a reason to consider `brew` special enough to mention it explicitly? Linux distros all come with some sort of a package manager and at least the rolling release distros keep up with the stable releases.

**Comment by henryiii** [comment]
Brew is the (by far) most popular macOS package manager, and it updates regularly (hours to days after releases). It also works (as Linuxbrew) on all flavors of linux, unlike system package managers. At least one pybind11 maintainer (me) is also a regular contributor to home-brew, and will help keep up this package if it falls behind. It's also mentioned in the original issue three years ago. :)

File: docs/installing.rst [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+Include as a submodule
+======================
+
+When you are working on a project in Git, you can use the pybind11 repository
+as a submodule. From your git repository, use:
+
+.. code-block:: bash
+
+    git submodule add ../../pybind/pybind11 extern/pybind11 -b v2.6.0
```
Comment (wjakob) [inline docs/installing.rst:19]: Cool (the relative link). Can we use `-b stable` if this gets baked into the docs?

**Review by wjakob — APPROVED** [review]
LGTM, just a minor comment.

File: docs/installing.rst (hunk 1 above)
Comment (henryiii) [reply docs/installing.rst:19]: Forgot about that, yes, that's better!

**Pull request merged**